OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectMatrix.cs

```diff
@@ -28,6 +29,82 @@ sealed class DreamMetaObjectMatrix : IDreamMetaObject {
             return array;
         }
 
+        /// <remarks>
+        /// This does elect to actually call /matrix/New() with the old matrix's values, <br/>
+        /// if anything, for the sake of support for having derived classes of /matrix.
+        /// </remarks>
+        /// <param name="matrix">The matrix to clone.</param>
+        /// <returns>A clone of the given matrix.</returns>
+        public static DreamObject MatrixClone(IDreamObjectTree ObjectTree, DreamObject matrix) {
+            var newMatrix = ObjectTree.CreateObject(matrix.ObjectDefinition.TreeEntry);
+            var args = new List<DreamValue>(6);
+            foreach(float f in EnumerateMatrix(matrix)) {
+                args.Add(new DreamValue(f));
+            }
+            newMatrix.InitSpawn(new Procs.DreamProcArguments(args));
+            return newMatrix;
+        }
+
+        /// <summary>
+        /// Simple helper for quickly making a basic matrix given its six values, in a-to-f order.
+        /// </summary>
+        /// <remarks>
+        /// Note that this skips over making a New() call, so hopefully you're not doing anything meaningful in there, DM-side.
```

Comment:
Why skip over `New()` in here but not `MatrixClone()`?

Reply:
This method is invoked in places where invoking a New() would require instantiating a new DreamThread, due to current native proc limitations.

I'll add a TODO about that, though.